code/game/objects/items/storage/holsters.dm

```diff
@@ -42,19 +45,25 @@
 
 /obj/item/storage/holster/handle_item_insertion(obj/item/W, prevent_warning = 0)
 	. = ..()
-	if (!. || !(W.type in holsterable_allowed) ) //check to see if the item being inserted is the snowflake item
+	if(!. || !is_type_in_list(W,holsterable_allowed)) //check to see if the item being inserted is the snowflake item
 		return
 	holstered_item = W
 	playsound(src, sheathe_sound, 15, 1)
-	update_icon()
+	if(istype(src, /obj/item/storage/holster/belt)) //Belt holsters have a pistol-in-belt sprite to overlay over the belt
```

Comment:
istype src is bad. Using the underlay should be a enabled by a flag or something, and be generic to all holsters.